src/main/kotlin/io/github/anki/anki/controller/CardsController.kt

```diff
@@ -16,21 +19,47 @@ import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
-@RequestMapping("/api/v1/cards")
+@RequestMapping("/api/v1/decks/{deckId}/cards")
 class CardsController(
-    private val service: CardsService,
+    private val cardService: CardsService,
 ) {
+    private val requestUserId = "66a11305dc669eefd22b5f3a"
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    fun createCard(@RequestBody request: NewCardRequest): CardDtoResponse {
-        val newCard: Card = service.createNewCard(request.toCard())
-        return newCard.toDto()
-    }
+    fun createCard(@Valid @RequestBody request: NewCardRequest, @PathVariable deckId: String): CardDtoResponse =
+        cardService.createNewCard(
+            userId = requestUserId,
+            request.toCard(deckId),
+        ).toDto()
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    fun getAllCardsFromDeck(@PathVariable deckId: String): List<CardDtoResponse> =
+        cardService.getAllCardsFromDeck(
+            deckId = deckId,
+            userId = requestUserId,
+        ).map { it.toDto() }
+
+    @PatchMapping("/{cardId}")
+    @ResponseStatus(HttpStatus.OK)
+    fun patchCard(
+        @PathVariable deckId: String,
```

Comment:
The `deckId ` not needed to update card by id

Reply:
but we need to verify that user has rights on this deck
also we have standartized api for card requests